fe/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -721,8 +728,12 @@ public void finishTransaction(long transactionId, Set<Long> errorReplicaIds) thr
                             }
 
                             if (healthReplicaNum < quorumReplicaNum) {
-                                LOG.info("publish version failed for transaction {} on tablet {},  with only {} replicas less than quorum {}",
+                                LOG.info("publish version failed for transaction {} on tablet {}, with only {} replicas less than quorum {}",
                                         transactionState, tablet, healthReplicaNum, quorumReplicaNum);
+                                String errMsg = String.format("publish on tablet %d failed. succeed replica num %d less than quorum %d."
+                                        + " table: %d, partition: %d, publish version: %d",
+                                        tablet.getId(), healthReplicaNum, quorumReplicaNum, tableId, partitionId, partition.getVisibleVersion() + 1);
+                                transactionState.setErrorMsg(errMsg);
```

Comment:
What is the different between version and lastSuccessVersion? Why only update lastSuccessVersion but not version when we found replica's lastFailedVersion is larger than 0?

Reply:
versions before `version` is  continuous. `lastSuccessVersion` is the last success version of this replica.  `lastFailedVersion` is the last failed version of this replica. eg:
versio: 10
lastSuccessVersion:12
lastFailedVersion:11

which means version 0-10 is continuous and success. and version 11 failed to publish. but 
version 12 success.